Homework 23.08.2024

HKV203 opened the pull request.

Commit:
The code that has variables userName, password, url in use

Commit:
org.postgresql.util.PSQLException: ERROR: column "avatar" of relation "users" does not exist

Commit:
ERROR: null value in column "id" of relation "users" violates not-null constraint
  Detail: Failing row contains (null, Twitter Admin, 2024-08-23, TwitterAdmin, null).] [insert into users (avatar,creation_date,name,username) values (?,?,?,?)]

Commit:
When I try to use values from application.properties

Unable to determine Dialect without JDBC metadata (please set 'jakarta.persistence.jdbc.url' for common cases or 'hibernate.dialect' when a custom Dialect implementation must be provided)

Commit:
Update

Commit:
Transactional update

Commit:
Transactional works now

Commit:
Homework 20.08.2024

Commit:
The first version

Commit:
The second version

VolgaDaniluk commented on src/main/java/app/controllers/TweetController.java:
```diff
@@ -0,0 +1,21 @@
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping(produces = "application/json")
```
you don't need it)

VolgaDaniluk commented on src/main/java/app/controllers/TweetController.java:
```diff
@@ -0,0 +1,21 @@
+public class TweetController {
+
+    @Autowired
+    TweetService service;
```
modifier is missing
better to use constructor injection

VolgaDaniluk commented on src/main/java/app/dao/UserDao.java:
```diff
@@ -0,0 +1,33 @@
+import org.springframework.stereotype.Component;
+
+@Component
+public class UserDao {
```
have to be a repository

VolgaDaniluk commented on src/main/java/app/hibernate/SessionFactoryProvider.java:
```diff
@@ -0,0 +1,25 @@
+import org.hibernate.boot.registry.StandardServiceRegistryBuilder;
+import org.hibernate.cfg.Configuration;
+
+public class SessionFactoryProvider {
```
don't need it

VolgaDaniluk commented on src/main/java/app/hibernate/hibernate.cfg.xml:
also don't need it

VolgaDaniluk commented on src/main/java/app/service/ContextConfiguration.java:
```diff
@@ -0,0 +1,36 @@
+@Configuration
+public class ContextConfiguration {
+
+    @Bean
+    public PlatformTransactionManager transactionManager(SessionFactory sessionFactory) {
+        HibernateTransactionManager transactionManager = new HibernateTransactionManager();
+        transactionManager.setSessionFactory(sessionFactory);
+        return transactionManager;
+    }
+
+    @Bean
+    public LocalSessionFactoryBean entityManagerFactory(DataSource dataSource) {
+        LocalSessionFactoryBean sessionFactory = new LocalSessionFactoryBean();
+        sessionFactory.setDataSource(dataSource);
+        sessionFactory.setPackagesToScan("app/entities");
+        return sessionFactory;
```
don't need it

Mopgonpblr replied:
When I try to delete entityManagerFactory I get this:

"
***************************
APPLICATION FAILED TO START
***************************

Description:

A component required a bean named 'entityManagerFactory' that could not be found.


Action:

Consider defining a bean named 'entityManagerFactory' in your configuration."

And when I try to delete transactionManager I get this: "Exception in thread "main" jakarta.persistence.TransactionRequiredException: no transaction is in progress"

VolgaDaniluk commented on src/main/java/app/service/TweetService.java:
```diff
@@ -0,0 +1,37 @@
+
+import java.util.List;
+
+@Component
+@Service
```
have to be only one of them

VolgaDaniluk commented on src/main/java/app/service/TweetService.java:
```diff
@@ -0,0 +1,37 @@
+@Service
+public class TweetService {
+
+    @Autowired
```
injection through constructor
name is incorrect

VolgaDaniluk commented on src/main/java/app/service/UserService.java:
same as for TweetService

VolgaDaniluk commented on pom.xml:
```diff
@@ -1,60 +1,66 @@
+            <version>6.5.2.Final</version>
+        </dependency>
 
+        <dependency>
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-core</artifactId>
+            <version>6.1.12</version>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-context</artifactId>
+            <version>6.1.12</version>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-autoconfigure</artifactId>
+            <version>3.3.3</version>
+        </dependency>
```
don't need this

VolgaDaniluk reviewed (commented):
where is the implementation of the task with security

HKV203 commented:
> where is the implementation of the task with security

<dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-security</artifactId>
            <version>3.3.3</version>
        </dependency>

in pom.xml

And 

spring.security.user.name=newuser
spring.security.user.password=newpassword2

in application.properties

Commit:
1. @RequestMapping in TweetController is deleted
2. Added constructors for TweetService in TweetController, TweetRepository in TweetService and UserDao in UserService
3. @Component is replaced with @Repository in UserDao
4. SessionFactoryProvider and hibernate.cfg.xml (for some reason didn't think about deleting them though they were excluded from the project before)
5. No more @Component in TweetService and UserService
6. pom.xml now doesn't have spring-core, spring-context and spring-boot-autoconfigure

Commit:
1. @RequestMapping in TweetController is deleted
2. Added constructors for TweetService in TweetController, TweetRepository in TweetService and UserDao in UserService
3. @Component is replaced with @Repository in UserDao
4. SessionFactoryProvider and hibernate.cfg.xml (for some reason didn't think about deleting them though they were excluded from the project before)
5. No more @Component in TweetService and UserService
6. pom.xml now doesn't have spring-core, spring-context and spring-boot-autoconfigure

Commit:
PlatformTransactionManager and LocalSessionFactoryBean beans are deleted
